examples/1_advanced/10_zkp_revocation.rs

```diff
@@ -0,0 +1,526 @@
+use identity_iota::storage::KeyType;
+use identity_iota::storage::TimeframeRevocationExtension;
+use identity_iota::verification::jws::JwsAlgorithm;
+use identity_iota::verification::MethodScope;
+use iota_sdk::client::secret::stronghold::StrongholdSecretManager;
+use iota_sdk::client::secret::SecretManager;
+use iota_sdk::client::Client;
+use iota_sdk::client::Password;
+use iota_sdk::types::block::address::Address;
+use iota_sdk::types::block::output::AliasOutput;
+use iota_sdk::types::block::output::AliasOutputBuilder;
+use iota_sdk::types::block::output::RentStructure;
+use jsonprooftoken::jpa::algs::ProofAlgorithm;
+use std::thread;
+use std::time::Duration as SleepDuration;
+
+async fn create_did(
+  client: &Client,
+  secret_manager: &SecretManager,
+  storage: &MemStorage,
+  key_type: KeyType,
+  alg: Option<JwsAlgorithm>,
+  proof_alg: Option<ProofAlgorithm>,
+) -> anyhow::Result<(Address, IotaDocument, String)> {
+  // Get an address with funds for testing.
+  let address: Address = get_address_with_funds(client, secret_manager, FAUCET_ENDPOINT).await?;
+
+  // Get the Bech32 human-readable part (HRP) of the network.
+  let network_name: NetworkName = client.network_name().await?;
+
+  // Create a new DID document with a placeholder DID.
+  // The DID will be derived from the Alias Id of the Alias Output after publishing.
+  let mut document: IotaDocument = IotaDocument::new(&network_name);
+
+  // New Verification Method containing a BBS+ key
+  let fragment = if let Some(alg) = alg {
+    document
+      .generate_method(storage, key_type, alg, None, MethodScope::VerificationMethod)
+      .await?
+  } else if let Some(proof_alg) = proof_alg {
+    let fragment = document
+      .generate_method_jwp(storage, key_type, proof_alg, None, MethodScope::VerificationMethod)
+      .await?;
+
+    // Create a new empty revocation bitmap. No credential is revoked yet.
+    let revocation_bitmap: RevocationBitmap = RevocationBitmap::new();
+
+    // Add the revocation bitmap to the DID document of the issuer as a service.
+    let service_id: DIDUrl = document.id().to_url().join("#my-revocation-service")?;
+    let service: Service = revocation_bitmap.to_service(service_id)?;
+
+    assert!(document.insert_service(service).is_ok());
+
+    fragment
+  } else {
+    return Err(anyhow::Error::msg("You have to pass at least one algorithm"));
+  };
+
+  // Construct an Alias Output containing the DID document, with the wallet address
+  // set as both the state controller and governor.
+  let alias_output: AliasOutput = client.new_did_output(address, document, None).await?;
+
+  // Publish the Alias Output and get the published DID document.
+  let document: IotaDocument = client.publish_did_output(secret_manager, alias_output).await?;
+  println!("Published DID document: {document:#}");
+
+  Ok((address, document, fragment))
+}
+
+/// Demonstrates how to create an Anonymous Credential with BBS+.
+#[tokio::main]
```

Comment:
This is needed to pause & advance time for the runtime later on
```suggestion
#[tokio::main(flavor = "current_thread")]
```